### ISSUE TITLE
Pasting individual list elements from Firefox on Windows fails

### ISSUE THREAD

**Issue by jackwickham** [ckeditor/ckeditor4#3415]
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Copy a fragment of a list using Firefox on Windows. For example, given a list with elements A, B, C and D, copying just B and C will produce clipboard contents
```html
<html><body>
<!--StartFragment--><li>
		B
	</li>
	<li>
		C
	</li><!--EndFragment-->
</body>
</html>
```
(per https://evercoder.github.io/clipboard-inspector/ as well as dev tool stepping through)

2. Paste this into a CKEditor instance

### Expected result

The paste succeeds, inserting

> * B
> * C

### Actual result

```
TypeError: maybeBlockData is undefined
```

## Root cause

This error appears to be caused because within the clipboard plugin, in [`_stripHtml`](https://github.com/ckeditor/ckeditor4/blob/9cb74b35db7082189088f47326eb05b02c947f8d/plugins/clipboard/plugin.js#L2730-L2750), the regexes that strip the above snippet trim the `<html>`, `<body>` and `<!--(Start|End)Fragment-->`, but they don't trim the whitespace between `<body>` and `<!--StartFragment-->`, and between `<!--EndFragment-->` and `</body>`. The trimmed result therefore has leading and trailing whitespace.

When this makes its way to core/editable.js, [`processDataForInsertion`](https://github.com/ckeditor/ckeditor4/blob/9cb74b35db7082189088f47326eb05b02c947f8d/core/editable.js#L1785), this triggers the leading/trailing whitespace check, so it is prepended and appended with a protection span, so it looks like
```html
<span data-cke-marker="1">&nbsp;</span>
<li>
		B
	</li>
	<li>
		C
	</li>
<span data-cke-marker="1">&nbsp;</span>
```

This is then parsed as HTML. As it stands it isn't valid HTML, because there are bare `<li>`s without a `<ul>`. When Firefox parses it, it fixes it up as well as it can, to produce HTML equivalent to
```html
<span data-cke-marker="1">&nbsp;</span>
<ul>
    <li>
        B
    </li>
    <li>
        C
    </li>
    <span data-cke-marker="1">&nbsp;</span>
</ul>
```
When you then trim the first and last elements of this, to drop the spans that are supposed to be leading and trailing, this deletes everything and causes the rest of the code to break.

## Other details

* Browser: Firefox 69 (I believe it only started occurring in 68 or 69)
* OS: Windows 10 (this does not occur in Linux)
* CKEditor version: 4.6.0
* Installed CKEditor plugins: clipboard

**Comment by Comandeer** [comment]
I can confirm that the issue is present only in Firefox and only on Windows.

**Comment by Comandeer** [comment]
It seems to be connected with most basic issue, #3520.

**Comment by Comandeer** [comment]
Closed in #3416.
